src/H5I.c

```diff
@@ -90,14 +90,13 @@ static int H5I__iterate_pub_cb(void *obj, hid_t id, void *udata);
  *-------------------------------------------------------------------------
  */
 H5I_type_t
-H5Iregister_type(size_t H5_ATTR_DEBUG_API_USED hash_size, unsigned reserved, H5I_free_t free_func)
+H5Iregister_type(size_t H5_ATTR_UNUSED hash_size, unsigned reserved, H5I_free_t free_func)
```

Comment:
Interesting. Very weird to have an unused parameter in an API function.

Reply:
It's because the original implementation of ID storage was a hash table that took a size parameter. It's an internal implementation detail that never should have been exposed. We should remove it and version the API call for 2.0.